### PR TITLE
Objects Automated Test - LPS-162184, LPS-162187, LPS-162188, LPS-162189 | master

### PR DESCRIPTION
**Pull request by andreefarias**
https://issues.liferay.com/browse/LPS-162184
https://issues.liferay.com/browse/LPS-162187
https://issues.liferay.com/browse/LPS-162188
https://issues.liferay.com/browse/LPS-162189

### CODE REVIEW

**Comment by andreefarias** [comment]
ci:test:sf

**Comment by andreefarias** [comment]
In the future, it's possible to refactor **CanUpdateCustomFieldInSystemObject** and **CanDeleteCustomFieldInSystemObject** with loops on the code.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 1]
```diff
@@ -815,6 +815,39 @@ definition {
...
+
+		task ("Select title field and assert it") {
+			CreateObject.selectTitleField(fieldLabel = "External Reference Code");
+
+			CreateObject.saveObject();
+
+			CreateObject.assertTitleFieldIsSelected(fieldLabel = "External Reference Code");
+		}
+	}
+
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:951]: @andreefarias how much effort would it be to assert the entries are using the ERC as a title field here? If it's complex, we can add it in another test

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 2]
```diff
@@ -615,6 +615,107 @@ definition {
...
+			ObjectPortlet.selectSystemObject(label = "Account");
+
+			ObjectAdmin.goToFieldsTab();
+		}
+
+		task ("Delete custom text field") {
+			ObjectAdmin.deleteObjectField(fieldName = "Custom Text");
+
+			ObjectAdmin.assertObjectFieldNotPresent(fieldLabel = "Custom Text");
+		}
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:646]: I just realized this too could be in a loop and the entire test could be two loops. However, I think this way is easier to read and follow along.

**Review by timpak — COMMENTED** [review]
Really great job with these tests, @andreefarias! I'm really glad you took the time to create different types of fields in `CanCreateCustomFieldInSystemObject` and that you are adding fields to every system object.

Can you take a look at my question for the ERC test? If it's doable in an hour or two, then go for it. If not, we can create a separate test for this.

**Comment by timpak** [comment]
One more thing, the story to add fields to system objects is behind a feature flag right now. You'll need to remove them on the tests for them to work. Can you add a commit to set the property? 

`property custom.properties = "feature.flag.LPS-135404=true${line.separator}feature.flag.LPS-159151=true";`

FYI, `custom.properties` allows us to set properties in portal-ext on CI

**Comment by andreefarias** [comment]
In peer review

**Comment by timpak** [comment]
Hey @andreefarias, can you fix the merge conflict? You can send them to BChan once you're done

**Pull request closed without merging**